internal/charm/charm.go

```diff
@@ -79,14 +75,11 @@ func MetaFormatReasons(ch CharmMeta) (Format, []FormatSelectionReason) {
 			reasons.Add(SelectionBases)
 		}
 	}
-	if len(ch.Meta().Series) > 0 {
-		reasons.Add(SelectionSeries)
-	}
 
 	// To be a format v1 you can have no series with no bases or containers, or
 	// just have a series slice.
 	format := FormatV1
-	if !reasons.Contains(SelectionSeries) && reasons.Intersection(formatV2Set).Size() > 0 {
+	if reasons.Intersection(formatV2Set).Size() > 0 {
 		format = FormatV2
 	}
```

Comment:
I need to think this one through. If we remove series, then if it doesn't contain bases (from a manifest) or containers it's a v1?

Do we support the deployment of local charms without a manifest? If that's a no, then I don't see the point of V1 charms?

Reply:
@jack-w-shaw can you prove that FormatV1 is never returned after this change during a deploy?

I also suspect the formatV2Set on the intersection is incorrect. When we call `MetaFormatReasons` we never add `SelectionContainers` to the set of reasons. So the intersection will end up being "does it have a base or not". If we do require a manifest for deployment, then will FormatV1 ever be used?

Reply:
I think you're right about SelectionContainers. Now add that to reasons if there are any containers in metadata

I have also added a test for this

Reply:
In terms of your other point, it seems originally a charm would be classified as FormatV1 if it has series no matter what.

To me this seems wrong (and we have already established there were problems here re containers). But oh well. This does bring into question how we would classify charms now we simply ignore the field. In my opinion, we should continue to ignore it, and optimistically hope a charm is v2 until proven otherwise (missing manifest)

But yes, as things stand, a charm having a manifest with at least 1 base => the charm is v2.

This then does mean that if we ban charms without manifests, then we also ban charms with v1 metadata

There is also the case of manifests without bases, but these are also invalid so we don;t need to worry.

Reply:
So if we're dropping support for charms without manifests, this does implicitly mean we're dropping support for v1 charms

As such, I believe that all this MetaFormat code is now redundant and can be removed. This would be a nice drop in complexity, even though most of the callers of CheckMeta or MetaFormat just error out if they return v1